tests/lib/tools/filter-debug-output

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python3
+
+import argparse
+import re
+import sys
+
+OPERATIONS = [
+    'Preparing', 'Executing', 'Restoring',
+    'Rebooting', 'Discarding', 'Allocating', 'Waiting',
+    'Allocated', 'Connecting', 'Connected', 'Sending',
+    'Error', 'Debug', 'WARNING'
+    ]
+
+def match_date(date):
+    return re.findall(r'\d{4}-\d{2}-\d{2}', date)
+
+def match_time(time):
+    return re.findall(r'\d{2}:\d{2}:\d{2}', time)
+
+# Debug line starts with 
+def match_debug(line):
+    parts = line.strip().split(' ')
+    return len(parts) > 5 and \
+        match_date(parts[0]) and \
+        match_time(parts[1]) and \
+        parts[2] == 'Debug' and \
+        parts[3] == 'output' and \
+        parts[4] == 'for'
+
+# Debug output finishes when a new other line starts
+def is_debug_finished(line):
+    parts = line.strip().split(' ')
+    return len(parts) > 3 and \
+        match_date(parts[0]) and \
+        match_time(parts[1]) and \
+        parts[2] in OPERATIONS
+
+
+# Process spread output lines
+def process_spread_output(debug_file):
+    debug_lines = ""
+    for line in sys.stdin:
+        if debug_lines:
+            if is_debug_finished(line):
+                with open(debug_file, 'a+') as myfile:
+                   myfile.write(debug_lines)
+
+                # There is a Debug output right after
+                if match_debug(line):
+                    debug_lines = line
+                else:
+                    print(line.strip())
+                    debug_lines = ""
+                continue
+            else:
+                debug_lines = debug_lines + line
```

Comment:
This will be somewhat inefficient even though python later on tried to optimize it. I think we could have a mode where we decide to print or not each line as we see it.

The ones we see should be printed more or less like this:

> line number:line text with right-hand-side spaces stripped

Reply:
to not accumulate lines you could do something like:
```
def process_debug(from_stream: io.TextIO, start_line: str, debug_count: int, debug_file) -> str:
    """Process lines from the start of the debug section until the last line,
    returns the first line right after the debug section
    """
    with open(debug_file, "a+") as outf:
        outf.write("..... some header")
        outf.write(start_line.strip())
        for line in from_stream:
            if is_debug_finished(line):
                outf.write(".... some footer")
                return line
            
            outf.write(line.strip())
        

def process_spread_output(debug_file):
    debug_num = 0
    for line in sys.stdin:
        if match_debug_start(line):
            debug_num += 1
            line = process_debug(sys.stdin, line, debug_num, debug_file)
        print(line.strip())

```

Reply:
implemented